applications/StructuralMechanicsApplication/python_scripts/common_methods_solvers.py

```diff
@@ -0,0 +1,152 @@
+    main_model_part_name = settings["model_part_name"].GetString()
+    current_process_info = model[main_model_part_name].ProcessInfo
+    existing_computation = current_process_info.Has(StructuralMechanicsApplication.EIGENVALUE_VECTOR)
+    structural_parts = ExtractModelParts(model, settings)
+    for part in structural_parts:
+        # Create auxiliar parameters
+        compute_damping_coefficients_settings = KratosMultiphysics.Parameters("""
+        {
+            "echo_level"          : 0,
+            "damping_ratio_0"     : 0.0,
+            "damping_ratio_1"     : -1.0,
+            "eigen_values_vector" : [0.0]
+        }
+        """)
+
+        # Setting custom parameters
+        compute_damping_coefficients_settings["echo_level"].SetInt(damping_settings["determine_rayleigh_damping_settings"]["echo_level"].GetInt())
+        compute_damping_coefficients_settings["damping_ratio_0"].SetDouble(damping_ratio_0)
+        compute_damping_coefficients_settings["damping_ratio_1"].SetDouble(damping_ratio_1)
+
+        # We check if the values are previously defined
+        properties = part.GetProperties()
+        for prop in properties:
+            if prop.Has(StructuralMechanicsApplication.SYSTEM_DAMPING_RATIO):
+                damping_settings["determine_rayleigh_damping_settings"]["damping_ratio_0"].SetDouble(prop.GetValue(StructuralMechanicsApplication.SYSTEM_DAMPING_RATIO))
+                break
+        for prop in properties:
+            if prop.Has(StructuralMechanicsApplication.SECOND_SYSTEM_DAMPING_RATIO):
+                damping_settings["determine_rayleigh_damping_settings"]["damping_ratio_1"].SetDouble(prop.GetValue(StructuralMechanicsApplication.SECOND_SYSTEM_DAMPING_RATIO))
+                break
+
+        # We have computed already the eigen values
+        current_process_info = part.ProcessInfo
+        precomputed_eigen_values = damping_settings["determine_rayleigh_damping_settings"]["eigen_values_vector"].GetVector()
+        if len(precomputed_eigen_values) > 1:
+            compute_damping_coefficients_settings["eigen_values_vector"].SetVector(precomputed_eigen_values)
+        else:
+            # If not computed eigen values already
+            if not existing_computation:
+                KratosMultiphysics.Logger.PrintInfo("::[MechanicalSolver]::", "EIGENVALUE_VECTOR not previously computed. Computing automatically, take care")
+                import eigen_solver_factory
+                eigen_linear_solver = eigen_solver_factory.ConstructSolver(damping_settings["determine_rayleigh_damping_settings"]["eigen_system_settings"])
+                builder_and_solver = KratosMultiphysics.ResidualBasedBlockBuilderAndSolverWithConstraints(eigen_linear_solver)
+                eigen_scheme = StructuralMechanicsApplication.EigensolverDynamicScheme()
+                eigen_solver = StructuralMechanicsApplication.EigensolverStrategy(part, eigen_scheme, builder_and_solver)
```

Comment:
How does this work?
If e.g. you take a middle-piece of a structure it won't work bcs there are no BCs defined on such a part

@AndreasWinterstein and I would say the flow of the program should be the following:
~~~py
if not processinfo.Has(RAYLEIGH_ALPHA) or RAYLEIGH_BETA # the user specifed sth globally
    if compute_damping_coefficients_settings["eigen_values_vector"].size()>1:
        # in this case the eigenvalues were calculated in a separate simulation, this is what we typically do
        eigenvalue_vector = compute_damping_coefficients_settings["eigen_values_vector"].GetVector()
    elif process_info.Has(StructuralMechanicsApplication.EIGENVALUE_VECTOR)
        # in this case the eigenvalues were calculated in a separate simulation right before this one
        eigenvalue_vector = process_info[StructuralMechanicsApplication.EIGENVALUE_VECTOR]
    else: # no previously existing computation, doing an automatic one:
        # perform eigenvalue -analysis for ENTIRE computing-mp
        ...
        eigenvalue_vector = process_info[StructuralMechanicsApplication.EIGENVALUE_VECTOR]

    for part in structural_parts:
        # check if none of the properties in this part have RAYLEIGH_ALPHA or RAYLEIGH_BETA
        # in this case the user specified sth and no automatic computation should be done

        # Then you can check if SYSTEM_DAMPING_RATIO or SECOND_SYSTEM_DAMPING_RATIO exist in those properties
        # if yes, then use those, if no, then use the globally defined ones

        # Then you can calculate RAYLEIGH_ALPHA and RAYLEIGH_BETA and put it in the Properties (not the ProcessInfo)
~~~

Reply:
we don't understand what is the point of doing a modal-analysis more than once...?

Reply:
You precisely complained about the fact that the damping coefficients are associated to different materials and different bodies, this is reason